apache/trafficcontrol
# TP: limits delivery service cloning to source ds type = target ds type

## What does this PR (Pull Request) do?

- [x] This PR fixes #3403 

because different ds "categories" (anymap, dns, http or steering) have different attributes, cloning a ds needs to be limited where source ds and target ds have types of the same category.

for example, you should not be able to clone a DNS* ds to an STEERING ds because that would add attributes to the steering ds that are not applicable (i.e. origin fqdn).

## Which Traffic Control components are affected by this PR?

- Traffic Portal

## What is the best way to verify this PR?
- navigate to a ds
- click More > Clone Delivery Service

you no longer have the option to choose the category of the target ds. Instead, the type of the source ds will be used to create the clone, however, that type can be changed but must be within a "category". Here are the categories and associated types:

DNS
- DNS
- DNS_LIVE
- DNS_LIVE_NATNL

HTTP
- HTTP
- HTTP_LIVE
- HTTP_LIVE_NATNL
- HTTP_NO_CACHE

ANY_MAP

STEERING
- STEERING
- CLIENT_STEERING

With this change you can no longer clone ds's across categories. You must stay within the category of the source ds.

## The following criteria are ALL met by this PR
